### PR TITLE
Introduce `HandleKindDataIsInvariant` helper

### PR DESCRIPTION
**Pull request by BruceForstall**
Small helper to identify if a handle kind is invariant.

### CODE REVIEW

**Comment by BruceForstall** [comment]
Extracted from https://github.com/dotnet/runtime/pull/94250

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -10796,6 +10794,27 @@ void GenTree::SetIndirExceptionFlags(Compiler* comp)
...
+bool GenTree::HandleKindDataIsInvariant(GenTreeFlags flags)
+{
+    GenTreeFlags handleKind = flags & GTF_ICON_HDL_MASK;
+    assert(handleKind != GTF_EMPTY);
+
+    // All handle types are assumed invariant except those specifically listed here.
+
+    return !((handleKind == GTF_ICON_STATIC_HDL) || // Pointer to a mutable class Static variable
+             (handleKind == GTF_ICON_BBC_PTR) ||    // Pointer to a mutable basic block count value
+             (handleKind == GTF_ICON_GLOBAL_PTR));  // Pointer to mutable data from the VM state
```
Comment (jakobbotsch) [inline src/coreclr/jit/gentree.cpp:10815]: De Morgan's this?

**Review by jakobbotsch — APPROVED** [review]
LGTM. Minor nit, feel free to ignore.

File: src/coreclr/jit/gentree.cpp (hunk 1 above)
Comment (filipnavara) [inline src/coreclr/jit/gentree.cpp:10815]:
```suggestion
    return (handleKind != GTF_ICON_STATIC_HDL) && // Pointer to a mutable class Static variable
           (handleKind != GTF_ICON_BBC_PTR) &&    // Pointer to a mutable basic block count value
           (handleKind != GTF_ICON_GLOBAL_PTR);  // Pointer to mutable data from the VM state
```

**Pull request merged**